From: Raj, Deepu <deepu.raj@ti.com>
Date: Thu, 3 May 2012 10:54:32 +0000 (+0530)
Subject: TI814X: Add support for detecting Cent PG3.0
X-Git-Url: http://arago-project.org/git/projects/?p=u-boot-omap3.git;a=commitdiff_plain;h=b590dff9490b32fbf184b8c37388f6f7936ce67f

TI814X: Add support for detecting Cent PG3.0

This patch enable support for detecting ti8148
version 3.0 silicon.

Signed-off-by: Raj, Deepu <deepu.raj@ti.com>
Signed-off-by: Nageswari Srinivasan <nageswari@ti.com>
---

diff --git a/arch/arm/cpu/arm_cortexa8/ti81xx/sys_info.c b/arch/arm/cpu/arm_cortexa8/ti81xx/sys_info.c
index d8f7d3a..141e51e 100644
--- a/arch/arm/cpu/arm_cortexa8/ti81xx/sys_info.c
+++ b/arch/arm/cpu/arm_cortexa8/ti81xx/sys_info.c
@@ -52,6 +52,8 @@ u32 get_cpu_rev(void)
 	 */
 	if (0x3 == rev || 0xc == rev)
 		return PG2_1;
+	else if (0x4 == rev)
+		return PG3_0;
 	else
 		return PG1_0;
 #endif
@@ -202,7 +204,8 @@ int print_cpuinfo (void)
 	}
 
 	if (rev < PG_END) {
-		char cpu_rev_str[5][4] = {"1.0", "1.1", "2.0", "2.1"}, *cpu_rev;
+		char cpu_rev_str[6][4] = {"1.0", "1.1", "2.0", "2.1", "3.0"};
+		char *cpu_rev;
 
 		cpu_rev = cpu_rev_str[rev];
 		printf("%s-%s rev %s\n",
diff --git a/arch/arm/include/asm/arch-ti81xx/sys_proto.h b/arch/arm/include/asm/arch-ti81xx/sys_proto.h
index 9d2e588..a031e9e 100644
--- a/arch/arm/include/asm/arch-ti81xx/sys_proto.h
+++ b/arch/arm/include/asm/arch-ti81xx/sys_proto.h
@@ -35,6 +35,7 @@ enum cpu_rev {
 	PG1_1 = 1,
 	PG2_0 = 2,
 	PG2_1 = 3,
+	PG3_0 = 4,
 	PG_END
 };
 
diff --git a/board/ti/ti8148/evm.c b/board/ti/ti8148/evm.c
index a254e79..6ae388b 100644
--- a/board/ti/ti8148/evm.c
+++ b/board/ti/ti8148/evm.c
@@ -102,7 +102,7 @@ int is_ddr3(void)
 	 * To use PG2.1 and DDR2 enable #define CONFIG_TI814X_EVM_DDR2
 	 * in "include/configs/ti8148_evm.h"
 	 */
-	if (PG2_1 == get_cpu_rev())
+	if (PG2_1 <= get_cpu_rev())
 		#ifdef CONFIG_TI814X_EVM_DDR2
 			return 0;
 		#else
@@ -157,7 +157,7 @@ int board_init(void)
 	/* setup RMII_REFCLK to be sourced from audio_pll */
 	__raw_writel(0x4, RMII_REFCLK_SRC);
 
-	if (PG2_1 == get_cpu_rev()) {
+	if (PG2_1 <= get_cpu_rev()) {
 		/*program GMII_SEL register for RGMII mode */
 		__raw_writel(0x30a,GMII_SEL);
 	}

